sched/queue/memory/simple.go

```diff
@@ -0,0 +1,81 @@
+package memory
+
+// In-memory Scheduler Queue
+
+import (
+	"fmt"
+	"github.com/scootdev/scoot/sched"
+	"github.com/scootdev/scoot/sched/queue"
+	"time"
+)
+
+func NewSimpleQueue() (queue.Queue, chan sched.Job) {
```

Comment:
Channels look weird at first, but they replace a lot of more confusing code involving sync primitives. They also let client code look more familiary.

For instance, how would you know when to peek? Maybe peek waits, but then peek wouldn't be able to be used in a select statement.

Now Dequeue is a method on the item that's returned.